enumerable.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # parent block method is the #each
+  # num_arr = Array(1..9)
+  # num_arr.each { |x| puts x.to_s }
+  def my_each(arr)
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index(arr)
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  def my_select(arr)
+    y = 0
+    new_arr = []
+    while y < arr.length
+      new_arr.push(arr[y]) if yield(arr[y])
+      y += 1
+    end
+    new_arr
+  end
+
+  def my_all?(arr)
+    w = 0
+    while w < arr.length
+      return false unless yield(arr[w])
+
+      w += 1
+    end
+    true
+  end
+
+  def any?(arr)
+    i = 0
+    while i < arr.length
+      return true if yield(arr[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?(arr)
+    i = 0
+    while i < arr.length
+      return false if yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count(arr)
+    i = 0
+    results = 0
+    while i < arr.length
+      results += 1 if yield(arr[i])
+      i += 1
+    end
+    results
+  end
+
+  def my_map(arr)
+    i = 0
+    new_arr = []
+    while i < arr.length
+      new_arr.push(arr[i]) if yield(arr[i])
+      i += 1
+    end
+    new_arr
+  end
+
+  def my_inject(arr, results)
+    i = 0
+    while i < arr.length
+      # variable reassignments => results
+      results = yield(results, arr[i])
+      i += 1
+    end
+    results
+  end
```

Comment:
- [x] Enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
`[1, 1, 1, 1, 1, 1, 1, 1].my_inject{proc { |sum, n| sum + n }}` should return `#<Proc:0x000055a055166cf0@main.rb:127>`

- [ ] Enumerables `my_inject`, when a symbol is specified, combines each element of the collection by applying the symbol as a named method
`[1,2,3,4,5].my_inject(&:+)` should return `15`